External/Plugins/ASCompletion/PluginMain.cs

```diff
@@ -446,104 +446,107 @@ public void HandleEvent(Object sender, NotifyEvent e, HandlingPriority priority)
-                            {
-                                ASComplete.OnShortcut(Keys.Control | Keys.Shift | Keys.Space, ASContext.CurSciControl);
-                                e.Handled = true;
-                            }
-                            else if (command == "ASCompletion.CtrlAltSpace")
-                            {
-                                ASComplete.OnShortcut(Keys.Control | Keys.Alt | Keys.Space, ASContext.CurSciControl);
-                                e.Handled = true;
-                            }
-                            else if (command == "ASCompletion.ContextualGenerator")
-                            {
-                                if (ASContext.HasContext && ASContext.Context.IsFileValid)
+                                // resolve element under cursor and send a CustomData event
+                                else if (command == "ASCompletion.ResolveElement")
+                                {
+                                    ASComplete.ResolveElement(sci, cmdData);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.MakeIntrinsic")
+                                {
+                                    ASContext.Context.MakeIntrinsic(cmdData);
+                                    e.Handled = true;
+                                }
+
+                                // alternative to default shortcuts
+                                else if (command == "ASCompletion.CtrlSpace")
+                                {
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.CtrlShiftSpace")
                                 {
-                                    ASGenerator.ContextualGenerator(ASContext.CurSciControl);
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Shift | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.CtrlAltSpace")
+                                {
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Alt | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.ContextualGenerator")
+                                {
+                                    if (ASContext.HasContext && ASContext.Context.IsFileValid)
+                                    {
+                                        ASGenerator.ContextualGenerator(ASContext.CurSciControl);
```

Comment:
I was halfway through doing that but then I realised there are 3.7 million `return` statements in the `ContextualGenerator` method, so I kinda went "nah cbb", which is a bad practice.
I will fix it in the next commit with the power of `Ctrl+H` however.